kotlin_spark/src/main/java/com/sleazyweasel/eboshi/ClientRoutes.kt

```diff
@@ -0,0 +1,22 @@
+package com.sleazyweasel.eboshi
+
+import org.eclipse.jetty.http.HttpStatus
+import spark.Response
+import javax.inject.Inject
+
+class ClientRoutes @Inject constructor(private val clientDataAccess: ClientDataAccess) {
+
+    val getAll = {
+        val clients = clientDataAccess.allClients()
+        JsonApiResponse(clients.map { it.toJsonApiObject() })
+    }
+
+    val create = { input: JsonApiRequest, response: Response ->
+        val client = Client(input.data.attributes)
```

Comment:
Its crazy how jealous I am of simple json destructuring after spending the last day in Haskell. :)